### PR TITLE
Video compression test2

### PR DESCRIPTION
**Pull request by BDSchoeler**
Added some video compression tests. The library can't run in normal testing rig because it is reliant c++ android libraries to run. Can't use power mock as a specific test runner has to be used.

### CODE REVIEW

File: test/androidTest/java/org/thoughtcrime/securesms/mms/VideoCompressionTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class VideoCompressionTest {
+    @Test(expected= IOException.class)
+    public void compressVideoTest() throws Exception{
```
Comment (benrs) [inline test/androidTest/java/org/thoughtcrime/securesms/mms/VideoCompressionTest.java:32]: Why exactly are we throwing an exception? Can we just try catch with a fail if an exception is caught?

File: test/androidTest/java/org/thoughtcrime/securesms/mms/VideoCompressionTest.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+        //act
+        String output = SiliCompressor.with(testContext).compressVideo(targetFile.getPath(), outputfolder);
+
+        //assert
+        assert( new File(output).length() < targetFile.length());
+    }
+
+    @Test(expected =  UnsupportedOperationException.class)
+    public void allowedAttachmentTest() throws Exception{
```
Comment (benrs) [inline test/androidTest/java/org/thoughtcrime/securesms/mms/VideoCompressionTest.java:52]: Same as above

Comment (BDSchoeler) [reply test/androidTest/java/org/thoughtcrime/securesms/mms/VideoCompressionTest.java:52]: It's the proper way of testing for if you want a method to throw an exception.

Comment (benrs) [reply test/androidTest/java/org/thoughtcrime/securesms/mms/VideoCompressionTest.java:52]: Do we expect an exception in both cases though? The above on seems to assert at the end.

Comment (BDSchoeler) [reply test/androidTest/java/org/thoughtcrime/securesms/mms/VideoCompressionTest.java:52]: Untill someone can figure out how to grant tests read/write permissions, yup.

Comment (claudds) [reply test/androidTest/java/org/thoughtcrime/securesms/mms/VideoCompressionTest.java:52]: Bryce is right. The test is expecting an exception to be thrown; that's the only way the test will actually pass when run & not throw us an actual error

**Review by claudds — APPROVED** [review]

File: test/androidTest/java/org/thoughtcrime/securesms/mms/VideoCompressionTest.java (hunk 2 above)
Comment (BDSchoeler) [reply test/androidTest/java/org/thoughtcrime/securesms/mms/VideoCompressionTest.java:52]: I've made issue #145 to address the inability for android/Instrumented unit tests.

Comment (lori-dalkin) [reply test/androidTest/java/org/thoughtcrime/securesms/mms/VideoCompressionTest.java:52]: I've read over the comments, with the bug issue created I'll approve this.

**Review by lori-dalkin — APPROVED** [review]

**Pull request merged**